packages/zowe-explorer/src/job/ZoweJobNode.ts

```diff
@@ -237,10 +237,14 @@ export class ZoweJobNode extends ZoweTreeNode implements IZoweJobTreeNode {
 
         const sortMethod = contextually.isSession(this) ? this.sort : { method: JobSortOpts.Id, direction: SortDirection.Ascending };
         // Remove any children that are no longer present in the built record
-        this.children = this.children
-            .concat(newChildren)
-            .filter((ch) => Object.values(elementChildren).find((recordCh) => recordCh.label === ch.label) != null)
-            .sort(ZoweJobNode.sortJobs(sortMethod));
+        this.children = contextually.isSession(this)
+            ? this.children
+                  .concat(newChildren)
+                  .filter((ch) => Object.values(elementChildren).find((recordCh) => recordCh.label === ch.label) != null)
```

Comment:
I see that lines 242-243 and 246-247 have the same logic. For readability, could this be reduced by checking if the node has a session before sorting the children array? For example:

```ts
this.children = this.children
    .concat(newChildren)
    .filter((ch) => Object.values(elementChildren).find((recordCh) => recordCh.label === ch.label) != null);
if (contextually.isSession(this)) {
    this.children.sort(ZoweJobNode.sortJobs(sortMethod));
}
```

Reply:
Earlier, I used the above-mentioned approach to escape code redundancy but then the SonarCloud Code Analysis - Maintainability check failed. So I have tried out this way @traeok .

Reply:
That's fair, I didn't realize that SonarCloud had flagged that section of code. Thanks for clarifying!